op-challenger/fault/solver.go

```diff
@@ -25,19 +25,56 @@ func NewSolver(gameDepth int, traceProvider TraceProvider) *Solver {
 func (s *Solver) NextMove(claim Claim) (*Claim, error) {
 	// Special case of the root claim
 	if claim.IsRoot() {
-		agree, err := s.agreeWithClaim(claim.ClaimData)
-		if err != nil {
-			return nil, err
-		}
-		// Attack the root claim if we do not agree with it
-		if !agree {
-			return s.attack(claim)
-		} else {
-			return nil, nil
-		}
+		return s.handleRoot(claim)
+	}
+	return s.handleMiddle(claim)
+}
+
+type StepData struct {
+	LeafClaim  Claim
+	StateClaim Claim
+	IsAttack   bool
+}
+
+func (s *Solver) AttemptStep(claim Claim, state Game) (StepData, error) {
+	if claim.Depth() != s.gameDepth {
+		return StepData{}, errors.New("cannot step on non-leaf claims")
```

Comment:
I think we should create a unit test around this path as well to make it explicit.

Reply:
done.